decentraland/unity-explorer
# feat: avatar shape sdk component

SDK7 Avatar Shape implementation.

* The world factory injects a `WorldProxy` to the `StaticContainer`, thus allowing to have a global world reference in scene world systems
* The new `AvatarShapeHandlerSystem` just forwards scene-created avatar shapes to the global world so that the already existent AvatarShape plugin takes care of rendering (running those systems per-scene would be a performance overkill). The system also takes care of cleaning up the global-world related entity when needed.
* In case anyone wants to see if working: I deployed a scene in prod at `60,-137` that has 2 avatar shapes instantiated by the scene, being repositioned every 5 seconds.

---------------------
Note: avatar shapes movement interpolation is out of the scope as that also has to work for other player avatars. In the old unity-renderer there's logic that affect any avatar shape (regardles of scene-created/player) interpolating their movement whenever their transform changes.

---------------------

## TEST STEPS FOR QA

1. Download the build artifact and open it
2. Use the teleport menu to go to `60,-137`
3. Connect to peer.decentraland.org realm
4. After everything loads confirm you see 2 avatars in that specific scene. Their position gets randomized every 5 seconds.
